[IA-3106] Github action to Create Vm Image

jdcanas opened the pull request:
Github action to Create GCE and Dataproc VM images via a daisy workflow

successful build here: https://github.com/DataBiosphere/leonardo/actions/runs/3943212134/jobs/6747789691 

(not triggered on most recent commits due to the goal for it be triggered manually)

Commit:
initial file creation with syntax tests

Commit:
script changes, debugging detect changes step

Commit:
dummy commit

Commit:
inputs section

Commit:
iterate

Commit:
base logic in place, untested

Commit:
google creds

Commit:
add push check to trigger for testing purposes

Commit:
syntax;

Commit:
comment out auto

Commit:
debug

mikeflinn commented on .github/workflows/custom_image_generation.yml:
```diff
@@ -0,0 +1,159 @@
+jobs:
+  run-image-generation-script-gce:
+    runs-on: ubuntu-latest
+    # needs: inspect-changes
```
```suggestion
    permissions:
      contents: 'read'
      id-token: 'write'
    # needs: inspect-changes
```

You need to add this block in order for workload id federation to work

mikeflinn commented on .github/workflows/custom_image_generation.yml:
```diff
@@ -0,0 +1,159 @@
+
+  run-image-generate-script-dataproc:
+    runs-on: ubuntu-latest
+    # needs: inspect-changes
```
```suggestion
    permissions:
      contents: 'read'
      id-token: 'write'
    # needs: inspect-changes
```
Same thing here

mikeflinn commented on .github/workflows/custom_image_generation.yml:
```diff
@@ -0,0 +1,159 @@
+          echo 'echoing test name'
+          echo "$TEST_NAME"
+          echo 'catting'
+          cat $GITHUB_WORKSPACE/jenkins/dataproc-custom-images/$GOOGLE_CREDENTIAL_FILE_NAME
```
Please never cat the contents of a service account file. Github actions logs are public, so any one who can see the leo repo could see this key. This should be done the same way as on line 74.

jdcanas replied:
ahh... this was just a sanity check, its not working currently, I just wanted to check if the file was indeed empty

mikeflinn commented on .github/workflows/custom_image_generation.yml:
```diff
@@ -0,0 +1,159 @@
+      - name: Run image generation script
+        run: |
+          docker run --rm -v $GITHUB_WORKSPACE/jenkins/dataproc-custom-images:/work:ro gcr.io/google.com/cloudsdktool/cloud-sdk:266.0.0 bash -c "cd /work/dataproc-custom-images \
+          && gcloud auth activate-service-account --key-file=/work/$GOOGLE_CREDENTIAL_FILE_NAME \
```
I think this needs to be the same file path you setup in line 126. I don't think there is actually anything at this `/work/$GOOGLE_CREDENTIAL_FILE_NAME` path. Maybe a copy paste from Jenkins?

jdcanas replied:
I'm fairly sure line 139 is mounting the directory with the key into the docker image's `work` dir. I'll remove the `:ro` though, not sure what that is.

mikeflinn replied:
Ah yeah I misread it, my bad. `:ro` means read-only volume

Commit:
feedback

Commit:
test

Commit:
test

Commit:
dir structure

Commit:
syntax

Commit:
fix gsutil?

mikeflinn commented on .github/workflows/custom_image_generation.yml:
```diff
@@ -0,0 +1,169 @@
+      - uses: actions/checkout@v3
+
+      - name: Create svc acct file
+        run: 'echo "$GCP_DAISY_GCR_KEY" > "$GITHUB_WORKSPACE/jenkins/dataproc-custom-images/$GOOGLE_CREDENTIAL_FILE_NAME"'
```
```suggestion
        run: 'echo "$GCP_DAISY_GCR_KEY" | base64 -d > "$GITHUB_WORKSPACE/jenkins/dataproc-custom-images/$GOOGLE_CREDENTIAL_FILE_NAME"'
```

Commit:
fix

Commit:
changes

Commit:
changes

Commit:
changes

Commit:
changes

Commit:
changes

Commit:
changes

Commit:
changes

Commit:
changes

Commit:
changes

Commit:
run script

Commit:
run script

Commit:
run script

Commit:
run script

Commit:
auth

Commit:
cleanup

Qi77Qi commented:
where can we find the generated image name?

also can we delete [test.tmp](https://github.com/DataBiosphere/leonardo/pull/3088/files#diff-1886643ed4b95f41725a642a0e1bc314b4b4c1658b74dfabc1b0227b05117c76)

Commit:
test

Commit:
syntax

Commit:
syntax

Commit:
syntax

Commit:
better output

Commit:
better output syntax

Commit:
better output syntax

Commit:
better output syntax

Commit:
better output syntax

Commit:
workingg last commit, better output summary

Commit:
simplify

Commit:
comments

jdcanas commented:
> where can we find the generated image name?
> 
> also can we delete [test.tmp](https://github.com/DataBiosphere/leonardo/pull/3088/files#diff-1886643ed4b95f41725a642a0e1bc314b4b4c1658b74dfabc1b0227b05117c76)

Deleted, and I added a step OUTPUT_IMAGE_URI that prints it. Unfortunately, there's not a good way to display it on the github actions UI that I found with my cursory googling..

jdcanas commented:
jenkins retest

Qi77Qi reviewed (approved):
looks great.. :shipit:

jdcanas commented:
jenkins retest
